### ISSUE TITLE
Revamp logic around offering of Kerbin contracts after a player goes interplanetary

### ISSUE THREAD

**Issue by jrossignol** [jrossignol/ContractPack-FieldResearch#10]
Currently the Kerbin contracts (most of them) stop being offered once a player starts to go interplanetary.  However, there's a few cases where this is not ideal:
1. A failed Mun free return could result in a probe being ejected from the system.
2. A player may simply want to do an easy local contract or may want to try to get all science.

So need to revisit how this is implemented with those in mind.

**Comment by TheDude044** [comment]
I like the idea of removing contracts, when player goes interplanetary. But is it possible to remove the trigger from entering the interplanetary space to entering the shpere of influence of any body outside the Kerbin's SOI. 
You don't need much to leave Kerbin's SOI, this could be done unintentionally, that doesn't mean, that you are ready to explore other planets. But if you, for example, entered the SOI of Duna or Eve, of course you did this with puprose, that would be detected as your readyness to explore other planets.

**Comment by jrossignol** [comment]
Agreed - that's where I was leaning for a fix if I decide to keep this logic.  I'm just not convinced that the logic is worth keeping (people can just ignore the offered contracts for Kerbin if they don't want them).  Or the middle ground is to reduce their frequency when the player goes interplanetary.
